### PR TITLE
build(core): remove jobs from pr and add to merge group.

### PR DESCRIPTION
**Pull request by mpaulucci**
**Motivation**
There are jobs that are quite unrelated to the files that have been changed, but still could cause an issue in an edge case. Let's run these tests on the merge queue instead of CI.

The main idea is that a job should be either in pull request or in merge group. Jobs on pull requests should be the jobs that test code that was changed and merge queues tests code that is probably unrelated but we want a sanity check before merging

**Description**
- Moved several heavy tests from CI (on pull request) to merge group
- Various formatting and consistency changes.
- Removed "ci skipped" hack since we already include those jobs in the merge queue

### CODE REVIEW

**Review by rodrigo-o — APPROVED** [review]
LGTM! I just left a small comment regarding using path filtering in `merge_group`. I think it could be great if @jrchatruc and @ilitteri give it also a look!

File: .github/workflows/ci_l2.yaml [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
   merge_group:
   pull_request:
     branches: ["**"]
-    paths-ignore:
-      - "README.md"
-      - "LICENSE"
-      - "**/README.md"
-      - "**/docs/**"
-      - "crates/vm/levm/**" # We ran this in a separate workflow
+    paths:
```
Comment (mpaulucci) [inline .github/workflows/ci_l2.yaml:6]: since it's added to merge groups, we still have a safety change for l1-only changes

File: .github/workflows/ci_skipped.yaml [hunk 2]
```diff
@@ -1,34 +0,0 @@
-name: CI Skipped
```
Comment (mpaulucci) [inline .github/workflows/ci_skipped.yaml:1]: Im going to remove required checks since they are part of the merge queue anyway, so a PR won't be merged without them

**Review by jrchatruc — APPROVED** [review]

File: .github/workflows/ci_levm.yaml [hunk 3]
```diff
@@ -2,10 +2,12 @@ name: CI LEVM
 
 on:
   merge_group:
+    paths:
+      - "crates/vm/levm/**"
```
[the remark this answers is not in the document]
Comment (mpaulucci) [reply .github/workflows/ci_levm.yaml:6]: will remove in a follow up
